contracts/DependencyRegistryV1.sol

```diff
@@ -0,0 +1,1074 @@
+     * @return dependencyType Dependency type used by project.
+     */
+    function getDependencyTypeForProject(
+        address _contractAddress,
+        uint256 _projectId
+    )
+        external
+        view
+        onlySupportedCoreContract(_contractAddress)
+        returns (string memory)
+    {
+        bytes32 dependencyType = projectDependencyTypeOverrides[
+            _contractAddress
+        ][_projectId];
+        if (dependencyType != bytes32(0)) {
+            return dependencyType.toString();
+        }
+
+        try
+            IDependencyRegistryCompatibleV0(_contractAddress)
+                .projectScriptDetails(_projectId)
+        returns (
+            string memory scriptTypeAndVersion,
+            string memory,
+            uint256 scriptCount
+        ) {
+            return scriptTypeAndVersion;
+        } catch {
+            revert(
+                "Contract does not implement projectScriptDetails and has no override set."
+            );
+        }
+    }
+
+    /**
+     * @notice Converts string to bytes32.
+     * https://ethereum.stackexchange.com/a/9152
+     * @param source string to be converted.
+     * @return result bytes32 representation of string.
+     */
+    function stringToBytes32(
+        string memory source
+    ) public pure returns (bytes32 result) {
+        bytes memory tempEmptyStringTest = bytes(source);
+        if (tempEmptyStringTest.length == 0) {
+            return 0x0;
+        }
+
+        assembly {
+            result := mload(add(source, 32))
+        }
+    }
+
+    /**
+     * @notice Converts bytes to hexadecimal string representation.
+     * https://ethereum.stackexchange.com/a/126928
+     * @param buffer bytes to be converted.
+     * @return string representation of buffer.
+     */
+    function bytesToHexString(
+        bytes memory buffer
+    ) public pure returns (string memory) {
+        // Fixed buffer size for hexadecimal convertion
+        bytes memory converted = new bytes(buffer.length * 2);
+
+        bytes memory _base = "0123456789abcdef";
+
+        for (uint256 i = 0; i < buffer.length; i++) {
+            converted[i * 2] = _base[uint8(buffer[i]) / _base.length];
+            converted[i * 2 + 1] = _base[uint8(buffer[i]) % _base.length];
+        }
+
+        return string(abi.encodePacked("0x", converted));
+    }
+
+    function getTokenHtml(
+        address _contractAddress,
+        uint256 _tokenId
+    ) external view returns (string memory) {
+        uint256 ONE_MILLION = 1_000_000;
+
+        uint256 projectId = _tokenId / ONE_MILLION;
+        bytes32 dependencyType = stringToBytes32(
+            this.getDependencyTypeForProject(_contractAddress, projectId)
+        );
+        bytes32 tokenHash = IDependencyRegistryCompatibleV0(_contractAddress)
+            .tokenIdToHash(_tokenId);
+
+        WrappedScriptRequest[] memory requests = new WrappedScriptRequest[](5);
+
+        requests[0]
```

Comment:
Note: Scripty.sol uses a single hardcoded html template so we use an extra script to setup our default template style. Alternatively we could fork the scripty contract and update the template there to include the style tag.

Reply:
I don't have super strong feelings here, but if we consider forking, perhaps @thedude would be willing to help figure out a more globally optimal solution here 👥

Reply:
I was actually planning to add another request type for the headers. Something like this:

HeaderTagRequest
- tagPrefix
- tagSuffix
- content

This way you can add custom tags in `<header></header >` tags. In your case you can define your css like this:
```
HeaderTagRequest[] memory headerTagRequests = new HeaderTagRequest[](1);
headerTagRequests[0].tagPrefix = "<style>";
headerTagRequests[0].tagPrefix = "</style>";
headerTagRequests[0].content = "html{height:100%}body{min-height:100%;margin:0;padding:0}canvas{padding:0;margin:auto;display:block;position:absolute;top:0;bottom:0;left:0;right:0}";
```

Maybe this way you can even add `<metadata>` to set description author title etc?
```
scriptyBuilder.getEncodedHTMLWrapped(
    headerTagRequests, 
    scriptRequests,
    bufferSize
);
```

scripty.sol is initially designed to focus on scripts but I think having this feature is valuable and makes scripty even more flexible. We wanted scripty.sol to be immutable so it has no owner and it's not upgradable but since it doesn't store anything and it has no dependency, deploying scriptyv2.sol shouldn't be a problem.

What do you think?

Reply:
Yes, I think this would be super useful and much cleaner! Just to be clear, you mean `<head></head>` tags and not header tags right?

Reply:
++ this is awesome and a nice way to handle the potential of supporting multiple templates (in the case we don't end up consolidating into a singular one)

Reply:
Yup I meant `<head></head>` tags :) alright I'll pick this up and let you know when I draft it.

Reply:
@0xdude if you haven't started on this yet, I'll get cracking on it tomo